buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchNode.java

```diff
@@ -289,47 +362,130 @@ private void waitForProcessToExit(ProcessHandle processHandle) {
         }
     }
 
-    private File getConfigFile() {
-        return new File(workingDir, "config/elasticsearch.yml");
-    }
-
-    private File getConfPathData() {
-        return new File(workingDir, "data");
-    }
-
-    private File getConfPathSharedData() {
-        return new File(workingDir, "sharedData");
-    }
-
-    private File getConfPathRepo() {
-        return new File(workingDir, "repo");
+    private void createWorkingDir(Path distroExtractDir) {
+        try {
+            syncWithLinks(distroExtractDir, workingDir);
+            Files.createDirectories(confPathRepo);
+            Files.createDirectories(confPathData);
+            Files.createDirectories(workingDir.resolve("sharedData"));
+            Files.createDirectories(confPathLogs);
+            Files.createDirectories(tmpDir);
+        } catch (IOException e) {
+            throw new UncheckedIOException("Failed to create working directory: " + workingDir, e);
+        }
     }
 
-    private File getConfPathLogs() {
-        return new File(workingDir, "logs");
+    public void syncWithLinks(Path sourceRoot, Path destinationRoot) throws IOException {
```

Comment:
Can you add a javadoc for this method? It is unclear to me why we need to add links. In the existing integ tests, we operate within the extracted dir.

Reply:
The previous implementation started the clusters without copying the extracted dir. 
This is no longer possible, since by installing plugins we are changing the plugins dir. 
A copy would have worked, this has a few advantages over that: 
  - faster, and uses less disk space   
  - Will use the same files from the kernel perspective, so chances of the jars being in FS cache are higher. 

By making the links read-only we make sure these are not accidentally written to and affect all test clusters mitigating the risk of the approach.
I think the complexity  is otherwise enclosed and not exposed to the users of the plugin.

Reply:
I'm not sure why plugins matter here. We would still be using the same config, data and log paths right? If we want to have a golden copy of the extracted dir, that is fine, but we can create separate dirs for plugins/config/data/logs and pass those in via cli params.

Reply:
I didn't find a way to customize the plugin directory, so plugins ended up installing in the golden copy and since we were already copying the parts of the config I taught it would  simplify things overall not to start the distribution from the golden copy but create it's own copy which we can set as cwd and use relative paths for better readability. Granted creating the links has complexity to it, but that's fairly local to the method that creates them.

Reply:
Sorry, I forgot we removed the ability to change the plugin directory. Let me ask a slightly different question from my original: is there a reason we could not continue to do a full extraction for each runner as we did before, and work on optimizations like this linking as separate followups? I am not convinced this is worth the complexity, as extraction should be a tiny fraction of the total time for all the integ tests.

Reply:
I would be ok with a project.sync in a followup to start. I'm not against sharing the extracted distribution in some way, but I would like to see it done as a standalone change that the implementation can be evaluated against other possible solutions, instead of bundled up with another change like this.